### ISSUE TITLE
System errors when running legacy code

### ISSUE THREAD

**Issue by iamlasse** [rectorphp/rector#7484]
[ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/get
         id3.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/classes/Admin.php
         " file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/classes/Core.php"
         file, due to:
         "System error: "Syntax error, unexpected '{':1"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 269

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/classes/Sync.php"
         file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/ext
         ension.cache.dbm.php" file, due to:
         "System error: "Invalid identifier name "''""
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 12

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/get
         id3.lib.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.archive.gzip.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.audio-video.asf.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.audio-video.matroska.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.audio-video.riff.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27
[ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.audio.aac.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/getid3/mod
         ule.audio.voc.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

 [ERROR] Could not process
         "/Users/lasselarsen/Development/wp_sites/testsite1/wordpress/wp-content/plugins/wp-filebase-8/extras/phpthumb.b
         mp.php" file, due to:
         "System error: "PHPStan\Type\Constant\ConstantIntegerType::__construct(): Argument #1 ($value) must be of type
         int, string given, called in
         phar://vendor/rector/rector/vendor/phpstan/phpstan/phpstan.phar/src/Reflection/InitializerExprTypeResolver.php:
         106"
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On
         line: 27

**Comment by samsonasik** [comment]
That’s should already solved in dev-main at https://github.com/rectorphp/rector-src/pull/2913, you can try:

```
composer config minimum-stability dev
composer config prefer-stable true
composer require --dev rector/rector:dev-main
```

**Comment by samsonasik** [comment]
Released 0.14.3 for it https://github.com/rectorphp/rector/releases/tag/0.14.3
